### ISSUE TITLE
The torch2.0 sample program reports an error when calling g++ on macos

### ISSUE THREAD

**Issue by crazy-JiangDongHua** [pytorch/pytorch#93196]
### 🐛 Describe the bug

My pytorch version is 2.0.0.dev20230127, MacOS version is 13.1, python version is 3.9.12. When I run the torch2.0 sample program in [here](https://pytorch.org/get-started/pytorch-2.0/#user-experience):
```
model = models.resnet18()
optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
compiled_model = torch.compile(model)
start = time.time()
x = torch.randn(16, 3, 224, 224)
optimizer.zero_grad()
out = compiled_model(x)
out.sum().backward()
optimizer.step()
end = time.time()
print(end-start)
```

### Error logs
```
CalledProcessError                        Traceback (most recent call last)
File /opt/miniconda3/lib/python3.9/site-packages/torch/_inductor/codecache.py:481, in CppCodeCache.load(cls, source_code)
    480 try:
--> 481     subprocess.check_output(cmd, stderr=subprocess.STDOUT)
    482 except subprocess.CalledProcessError as e:

File /opt/miniconda3/lib/python3.9/subprocess.py:424, in check_output(timeout, *popenargs, **kwargs)
    422     kwargs['input'] = empty
--> 424 return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
    425            **kwargs).stdout

File /opt/miniconda3/lib/python3.9/subprocess.py:528, in run(input, capture_output, timeout, check, *popenargs, **kwargs)
    527     if check and retcode:
--> 528         raise CalledProcessError(retcode, process.args,
    529                                  output=stdout, stderr=stderr)
    530 return CompletedProcess(process.args, retcode, stdout, stderr)

CalledProcessError: Command '['g++', '/var/folders/x7/scywf31j5znbyrc7sx9fyd900000gn/T/torchinductor_frankj/nv/cnvvffjhbnznyauxn6jlzj5qjcgxza6svl7hi5kmnnallfw3xfu4.cpp', '-shared', '-fPIC', '-Wall', '-std=c++17', '-Wno-unused-variable', '-I/opt/miniconda3/lib/python3.9/site-packages/torch/include', '-I/opt/miniconda3/lib/python3.9/site-packages/torch/include/torch/csrc/api/include', '-I/opt/miniconda3/lib/python3.9/site-packages/torch/include/TH', '-I/opt/miniconda3/lib/python3.9/site-packages/torch/include/THC', '-I/opt/miniconda3/include/python3.9', '-lgomp', '-march=native', '-O3', '-ffast-math', '-fno-finite-math-only', '-fopenmp', '-D', 'C10_USING_CUSTOM_GENERATED_MACROS', '-o/var/folders/x7/scywf31j5znbyrc7sx9fyd900000gn/T/torchinductor_frankj/nv/cnvvffjhbnznyauxn6jlzj5qjcgxza6svl7hi5kmnnallfw3xfu4.so']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

CppCompileError                           Traceback (most recent call last)
File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/output_graph.py:678, in OutputGraph.call_user_compiler(self, gm)
    677 else:
...


You can suppress this exception and fall back to eager by setting:
    torch._dynamo.config.suppress_errors = True
```

### Minified repro
```
import torch
import torch._dynamo as dynama
import torchvision.models as models
import time
```

### Versions

when I run collect_env.py, it report a error:
```
Collecting environment information...
Traceback (most recent call last):
  File "/Users/frankj/Documents/AI-System/collect_env.py", line 505, in <module>
    main()
  File "/Users/frankj/Documents/AI-System/collect_env.py", line 488, in main
    output = get_pretty_env_info()
  File "/Users/frankj/Documents/AI-System/collect_env.py", line 483, in get_pretty_env_info
    return pretty_str(get_env_info())
  File "/Users/frankj/Documents/AI-System/collect_env.py", line 330, in get_env_info
    pip_version, pip_list_output = get_pip_packages(run_lambda)
  File "/Users/frankj/Documents/AI-System/collect_env.py", line 302, in get_pip_packages
    out = run_with_pip(sys.executable + ' -mpip')
  File "/Users/frankj/Documents/AI-System/collect_env.py", line 290, in run_with_pip
    for line in out.splitlines()
AttributeError: 'NoneType' object has no attribute 'splitlines'
(base) frankj@frankjdeMacBook-Pro AI-System % python3 collect_env.py
Collecting environment information...
Traceback (most recent call last):
  File "/Users/frankj/Documents/AI-System/collect_env.py", line 505, in <module>
    main()
  File "/Users/frankj/Documents/AI-System/collect_env.py", line 488, in main
    output = get_pretty_env_info()
  File "/Users/frankj/Documents/AI-System/collect_env.py", line 483, in get_pretty_env_info
    return pretty_str(get_env_info())
  File "/Users/frankj/Documents/AI-System/collect_env.py", line 330, in get_env_info
    pip_version, pip_list_output = get_pip_packages(run_lambda)
  File "/Users/frankj/Documents/AI-System/collect_env.py", line 302, in get_pip_packages
    out = run_with_pip(sys.executable + ' -mpip')
  File "/Users/frankj/Documents/AI-System/collect_env.py", line 290, in run_with_pip
    for line in out.splitlines()
AttributeError: 'NoneType' object has no attribute 'splitlines'
```
cc @malfet @albanD @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh

**Comment by ezyang** [comment]
You have truncated your error logs but I am expecting to see some message from your c++ compiler. Are you able to post the entirety of your logs to gist or something?

**Comment by ezyang** [comment]
@malfet this is probably the thing where the cpu backend doesn't work on osx right now

**Comment by crazy-JiangDongHua** [comment]
Oh, I see, the real error is
```
clang: error: the clang compiler does not support '-march=native'
clang: error: unsupported option '-fopenmp'
clang: error: unsupported option '-fopenmp'
```

and the full error log is:
```
---------------------------------------------------------------------------
CalledProcessError                        Traceback (most recent call last)
File /opt/miniconda3/lib/python3.9/site-packages/torch/_inductor/codecache.py:481, in CppCodeCache.load(cls, source_code)
    480 try:
--> 481     subprocess.check_output(cmd, stderr=subprocess.STDOUT)
    482 except subprocess.CalledProcessError as e:

File /opt/miniconda3/lib/python3.9/subprocess.py:424, in check_output(timeout, *popenargs, **kwargs)
    422     kwargs['input'] = empty
--> 424 return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
    425            **kwargs).stdout

File /opt/miniconda3/lib/python3.9/subprocess.py:528, in run(input, capture_output, timeout, check, *popenargs, **kwargs)
    527     if check and retcode:
--> 528         raise CalledProcessError(retcode, process.args,
    529                                  output=stdout, stderr=stderr)
    530 return CompletedProcess(process.args, retcode, stdout, stderr)

CalledProcessError: Command '['g++', '/var/folders/x7/scywf31j5znbyrc7sx9fyd900000gn/T/torchinductor_frankj/nv/cnvvffjhbnznyauxn6jlzj5qjcgxza6svl7hi5kmnnallfw3xfu4.cpp', '-shared', '-fPIC', '-Wall', '-std=c++17', '-Wno-unused-variable', '-I/opt/miniconda3/lib/python3.9/site-packages/torch/include', '-I/opt/miniconda3/lib/python3.9/site-packages/torch/include/torch/csrc/api/include', '-I/opt/miniconda3/lib/python3.9/site-packages/torch/include/TH', '-I/opt/miniconda3/lib/python3.9/site-packages/torch/include/THC', '-I/opt/miniconda3/include/python3.9', '-lgomp', '-march=native', '-O3', '-ffast-math', '-fno-finite-math-only', '-fopenmp', '-D', 'C10_USING_CUSTOM_GENERATED_MACROS', '-o/var/folders/x7/scywf31j5znbyrc7sx9fyd900000gn/T/torchinductor_frankj/nv/cnvvffjhbnznyauxn6jlzj5qjcgxza6svl7hi5kmnnallfw3xfu4.so']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

CppCompileError                           Traceback (most recent call last)
File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/output_graph.py:678, in OutputGraph.call_user_compiler(self, gm)
    677 else:
--> 678     compiled_fn = compiler_fn(gm, self.fake_example_inputs())
    679 _step_logger()(logging.INFO, f"done compiler function {name}")

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/debug_utils.py:1047, in wrap_backend_debug.<locals>.debug_wrapper(gm, example_inputs, **kwargs)
   1046 else:
-> 1047     compiled_gm = compiler_fn(gm, example_inputs, **kwargs)
   1049 return compiled_gm

File /opt/miniconda3/lib/python3.9/site-packages/torch/__init__.py:1331, in _TorchCompileInductorWrapper.__call__(self, model_, inputs_)
   1330 with self.cm:
-> 1331     return self.compile_fn(model_, inputs_)

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/optimizations/backends.py:24, in register_backend.<locals>.inner(gm, example_inputs, **kwargs)
     22 @functools.wraps(fn)
     23 def inner(gm, example_inputs, **kwargs):
---> 24     return fn(gm, example_inputs, **kwargs)

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/optimizations/backends.py:61, in inductor(*args, **kwargs)
     59 from torch._inductor.compile_fx import compile_fx
---> 61 return compile_fx(*args, **kwargs)

File /opt/miniconda3/lib/python3.9/site-packages/torch/_inductor/compile_fx.py:413, in compile_fx(model_, example_inputs_, inner_compile)
    408 with overrides.patch_functions():
    409 
    410     # TODO: can add logging before/after the call to create_aot_dispatcher_function
    411     # in torch._functorch/aot_autograd.py::aot_module_simplified::aot_function_simplified::new_func
    412     # once torchdynamo is merged into pytorch
--> 413     return aot_autograd(
    414         fw_compiler=fw_compiler,
    415         bw_compiler=bw_compiler,
    416         decompositions=select_decomp_table(),
    417         partition_fn=functools.partial(
    418             min_cut_rematerialization_partition, compiler="inductor"
    419         ),
    420     )(model_, example_inputs_)

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/optimizations/training.py:74, in aot_autograd.<locals>.compiler_fn(gm, example_inputs)
     73 with enable_aot_logging():
---> 74     cg = aot_module_simplified(gm, example_inputs, **kwargs)
     75     counters["aot_autograd"]["ok"] += 1

File /opt/miniconda3/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py:2475, in aot_module_simplified(mod, args, fw_compiler, bw_compiler, partition_fn, decompositions, hasher_type, static_argnums)
   2473 full_args.extend(args)
-> 2475 compiled_fn = create_aot_dispatcher_function(
   2476     functional_call,
   2477     full_args,
   2478     aot_config,
   2479 )
   2481 # TODO: There is something deeply wrong here; compiled_fn running with
   2482 # the boxed calling convention, but aot_module_simplified somehow
   2483 # historically returned a function that was not the boxed calling
   2484 # convention.  This should get fixed...

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/utils.py:160, in dynamo_timed.<locals>.dynamo_timed_inner.<locals>.time_wrapper(*args, **kwargs)
    159 t0 = time.time()
--> 160 r = func(*args, **kwargs)
    161 time_spent = time.time() - t0

File /opt/miniconda3/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py:2172, in create_aot_dispatcher_function(flat_fn, flat_args, aot_config)
   2170 # You can put more passes here
-> 2172 compiled_fn = compiler_fn(flat_fn, fake_flat_args, aot_config)
   2174 if not hasattr(compiled_fn, "_boxed_call"):

File /opt/miniconda3/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py:1411, in aot_wrapper_dedupe(flat_fn, flat_args, aot_config, compiler_fn)
   1410     if ok:
-> 1411         return compiler_fn(flat_fn, leaf_flat_args, aot_config)
   1413 # Strategy 2: Duplicate specialize.
   1414 #
   1415 # In Haskell types, suppose you have:
   (...)
   1447 #   }
   1448 #   keep_arg_mask = [True, True, False, True]

File /opt/miniconda3/lib/python3.9/site-packages/torch/_functorch/aot_autograd.py:1680, in aot_dispatch_autograd(flat_fn, flat_args, aot_config)
   1679     with track_graph_compiling(aot_config, "forward"):
-> 1680         compiled_fw_func = aot_config.fw_compiler(
   1681             fw_module, flat_args_with_views_handled
   1682         )
   1684 class CompiledFunction(torch.autograd.Function):

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/utils.py:160, in dynamo_timed.<locals>.dynamo_timed_inner.<locals>.time_wrapper(*args, **kwargs)
    159 t0 = time.time()
--> 160 r = func(*args, **kwargs)
    161 time_spent = time.time() - t0

File /opt/miniconda3/lib/python3.9/site-packages/torch/_inductor/compile_fx.py:388, in compile_fx.<locals>.fw_compiler(model, example_inputs)
    387 fixed = len(example_inputs) - num_example_inputs
--> 388 return inner_compile(
    389     model,
    390     example_inputs,
    391     num_fixed=fixed,
    392     cudagraphs=cudagraphs,
    393     graph_id=graph_id,
    394 )

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/debug_utils.py:586, in wrap_compiler_debug.<locals>.debug_wrapper(gm, example_inputs, **kwargs)
    585 else:
--> 586     compiled_fn = compiler_fn(gm, example_inputs, **kwargs)
    588 return compiled_fn

File /opt/miniconda3/lib/python3.9/site-packages/torch/_inductor/debug.py:239, in DebugContext.wrap.<locals>.inner(*args, **kwargs)
    238 with DebugContext():
--> 239     return fn(*args, **kwargs)

File /opt/miniconda3/lib/python3.9/contextlib.py:79, in ContextDecorator.__call__.<locals>.inner(*args, **kwds)
     78 with self._recreate_cm():
---> 79     return func(*args, **kwds)

File /opt/miniconda3/lib/python3.9/site-packages/torch/_inductor/compile_fx.py:151, in compile_fx_inner(gm, example_inputs, cudagraphs, num_fixed, is_backward, graph_id)
    150     graph.run(*example_inputs)
--> 151     compiled_fn = graph.compile_to_fn()
    153 if cudagraphs:

File /opt/miniconda3/lib/python3.9/site-packages/torch/_inductor/graph.py:560, in GraphLowering.compile_to_fn(self)
    559 def compile_to_fn(self):
--> 560     return self.compile_to_module().call

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/utils.py:160, in dynamo_timed.<locals>.dynamo_timed_inner.<locals>.time_wrapper(*args, **kwargs)
    159 t0 = time.time()
--> 160 r = func(*args, **kwargs)
    161 time_spent = time.time() - t0

File /opt/miniconda3/lib/python3.9/site-packages/torch/_inductor/graph.py:549, in GraphLowering.compile_to_module(self)
    547     print(code)
--> 549 mod = PyCodeCache.load(code)
    550 for name, value in self.constants.items():

File /opt/miniconda3/lib/python3.9/site-packages/torch/_inductor/codecache.py:504, in PyCodeCache.load(cls, source_code)
    503 mod.key = key
--> 504 exec(code, mod.__dict__, mod.__dict__)
    505 # another thread might set this first

File /var/folders/x7/scywf31j5znbyrc7sx9fyd900000gn/T/torchinductor_frankj/uw/cuwxp2sfqv3k6dy4etczd3ofbpkganasn2eo4l57y7rdwcrpi2bc.py:3345
   3103 kernel_cpp_20 = async_compile.cpp('''
   3104 #include "/var/folders/x7/scywf31j5znbyrc7sx9fyd900000gn/T/torchinductor_frankj/77/c7773nj5pwikpmm2pwa62rcudlf7p3if7eyqb5k4sjsvewwje4le.h"
   3105 extern "C" void kernel(const float* __restrict__ in_ptr0,
   (...)
   3341 }
   3342 ''')
-> 3345 async_compile.wait(globals())
   3346 del async_compile

File /opt/miniconda3/lib/python3.9/site-packages/torch/_inductor/codecache.py:699, in AsyncCompile.wait(self, scope)
    698 if isinstance(result, (Future, TritonFuture)):
--> 699     scope[key] = result.result()
    700     pbar.update(1)

File /opt/miniconda3/lib/python3.9/concurrent/futures/_base.py:446, in Future.result(self, timeout)
    445 elif self._state == FINISHED:
--> 446     return self.__get_result()
    447 else:

File /opt/miniconda3/lib/python3.9/concurrent/futures/_base.py:391, in Future.__get_result(self)
    390 try:
--> 391     raise self._exception
    392 finally:
    393     # Break a reference cycle with the exception in self._exception

File /opt/miniconda3/lib/python3.9/concurrent/futures/thread.py:58, in _WorkItem.run(self)
     57 try:
---> 58     result = self.fn(*self.args, **self.kwargs)
     59 except BaseException as exc:

File /opt/miniconda3/lib/python3.9/site-packages/torch/_inductor/codecache.py:676, in AsyncCompile.cpp.<locals>.task()
    675 def task():
--> 676     return CppCodeCache.load(source_code).kernel

File /opt/miniconda3/lib/python3.9/site-packages/torch/_inductor/codecache.py:483, in CppCodeCache.load(cls, source_code)
    482     except subprocess.CalledProcessError as e:
--> 483         raise exc.CppCompileError(cmd, e.output) from e
    485 cls.cache[key] = cls._load_library(output_path)

CppCompileError: C++ compile error

Command:
g++ /var/folders/x7/scywf31j5znbyrc7sx9fyd900000gn/T/torchinductor_frankj/nv/cnvvffjhbnznyauxn6jlzj5qjcgxza6svl7hi5kmnnallfw3xfu4.cpp -shared -fPIC -Wall -std=c++17 -Wno-unused-variable -I/opt/miniconda3/lib/python3.9/site-packages/torch/include -I/opt/miniconda3/lib/python3.9/site-packages/torch/include/torch/csrc/api/include -I/opt/miniconda3/lib/python3.9/site-packages/torch/include/TH -I/opt/miniconda3/lib/python3.9/site-packages/torch/include/THC -I/opt/miniconda3/include/python3.9 -lgomp -march=native -O3 -ffast-math -fno-finite-math-only -fopenmp -D C10_USING_CUSTOM_GENERATED_MACROS -o/var/folders/x7/scywf31j5znbyrc7sx9fyd900000gn/T/torchinductor_frankj/nv/cnvvffjhbnznyauxn6jlzj5qjcgxza6svl7hi5kmnnallfw3xfu4.so

Output:
clang: error: the clang compiler does not support '-march=native'
clang: error: unsupported option '-fopenmp'
clang: error: unsupported option '-fopenmp'


The above exception was the direct cause of the following exception:

BackendCompilerFailed                     Traceback (most recent call last)
Cell In[5], line 4
      2 x = torch.randn(16, 3, 224, 224)
      3 optimizer.zero_grad()
----> 4 out = compiled_model(x)

File /opt/miniconda3/lib/python3.9/site-packages/torch/nn/modules/module.py:1488, in Module._call_impl(self, *args, **kwargs)
   1483 # If we don't have any hooks, we want to skip the rest of the logic in
   1484 # this function, and just call forward.
   1485 if not (self._backward_hooks or self._backward_pre_hooks or self._forward_hooks or self._forward_pre_hooks
   1486         or _global_backward_pre_hooks or _global_backward_hooks
   1487         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1488     return forward_call(*args, **kwargs)
   1489 # Do not call functions when jit is used
   1490 full_backward_hooks, non_full_backward_hooks = [], []

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/eval_frame.py:82, in OptimizedModule.forward(self, *args, **kwargs)
     81 def forward(self, *args, **kwargs):
---> 82     return self.dynamo_ctx(self._orig_mod.forward)(*args, **kwargs)

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/eval_frame.py:211, in _TorchDynamoContext.__call__.<locals>._fn(*args, **kwargs)
    209 dynamic_ctx.__enter__()
    210 try:
--> 211     return fn(*args, **kwargs)
    212 finally:
    213     set_eval_frame(prior)

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/eval_frame.py:332, in catch_errors_wrapper.<locals>.catch_errors(frame, cache_size)
    329             return hijacked_callback(frame, cache_size, hooks)
    331 with compile_lock:
--> 332     return callback(frame, cache_size, hooks)

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:403, in convert_frame.<locals>._convert_frame(frame, cache_size, hooks)
    401 counters["frames"]["total"] += 1
    402 try:
--> 403     result = inner_convert(frame, cache_size, hooks)
    404     counters["frames"]["ok"] += 1
    405     return result

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:103, in wrap_convert_context.<locals>._fn(*args, **kwargs)
    101 torch.fx.graph_module._forward_from_src = fx_forward_from_src_skip_result
    102 try:
--> 103     return fn(*args, **kwargs)
    104 finally:
    105     torch._C._set_grad_enabled(prior_grad_mode)

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:261, in convert_frame_assert.<locals>._convert_frame_assert(frame, cache_size, hooks)
    258 global initial_grad_state
    259 initial_grad_state = torch.is_grad_enabled()
--> 261 return _compile(
    262     frame.f_code,
    263     frame.f_globals,
    264     frame.f_locals,
    265     frame.f_builtins,
    266     compiler_fn,
    267     one_graph,
    268     export,
    269     hooks,
    270     frame,
    271 )

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/utils.py:160, in dynamo_timed.<locals>.dynamo_timed_inner.<locals>.time_wrapper(*args, **kwargs)
    158     compilation_metrics[key] = []
    159 t0 = time.time()
--> 160 r = func(*args, **kwargs)
    161 time_spent = time.time() - t0
    162 # print(f"Dynamo timer: key={key}, latency={latency:.2f} sec")

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:323, in _compile(code, globals, locals, builtins, compiler_fn, one_graph, export, hooks, frame)
    321 for attempt in itertools.count():
    322     try:
--> 323         out_code = transform_code_object(code, transform)
    324         orig_code_map[out_code] = code
    325         break

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/bytecode_transformation.py:341, in transform_code_object(code, transformations, safe)
    338 instructions = cleaned_instructions(code, safe)
    339 propagate_line_nums(instructions)
--> 341 transformations(instructions, code_options)
    343 fix_vars(instructions, code_options)
    345 dirty = True

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/convert_frame.py:310, in _compile.<locals>.transform(instructions, code_options)
    297 nonlocal output
    298 tracer = InstructionTranslator(
    299     instructions,
    300     code,
   (...)
    308     mutated_closure_cell_contents,
    309 )
--> 310 tracer.run()
    311 output = tracer.output
    312 assert output is not None

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py:1717, in InstructionTranslator.run(self)
   1715 def run(self):
   1716     _step_logger()(logging.INFO, f"torchdynamo start tracing {self.f_code.co_name}")
-> 1717     super().run()

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py:563, in InstructionTranslatorBase.run(self)
    558 try:
    559     self.output.push_tx(self)
    560     while (
    561         self.instruction_pointer is not None
    562         and not self.output.should_exit
--> 563         and self.step()
    564     ):
    565         pass
    566 except BackendCompilerFailed:

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py:526, in InstructionTranslatorBase.step(self)
    524     if not hasattr(self, inst.opname):
    525         unimplemented(f"missing: {inst.opname}")
--> 526     getattr(self, inst.opname)(inst)
    528     return inst.opname != "RETURN_VALUE"
    529 except BackendCompilerFailed:

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/symbolic_convert.py:1783, in InstructionTranslator.RETURN_VALUE(self, inst)
   1778 _step_logger()(
   1779     logging.INFO,
   1780     f"torchdynamo done tracing {self.f_code.co_name} (RETURN_VALUE)",
   1781 )
   1782 log.debug("RETURN_VALUE triggered compile")
-> 1783 self.output.compile_subgraph(self)
   1784 self.output.add_output_instructions([create_instruction("RETURN_VALUE")])

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/output_graph.py:525, in OutputGraph.compile_subgraph(self, tx, partial_convert, reason)
    511     self.add_output_instructions(random_calls_instructions)
    513 if (
    514     stack_values
    515     and all(
   (...)
    522 
    523     # optimization to generate better code in a common case
    524     self.add_output_instructions(
--> 525         self.compile_and_call_fx_graph(tx, list(reversed(stack_values)), root)
    526         + [create_instruction("UNPACK_SEQUENCE", len(stack_values))]
    527     )
    528 else:
    529     graph_output_var = self.new_var("graph_out")

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/output_graph.py:596, in OutputGraph.compile_and_call_fx_graph(self, tx, rv, root)
    594 assert_no_fake_params_or_buffers(gm)
    595 with tracing(self.tracing_context):
--> 596     compiled_fn = self.call_user_compiler(gm)
    597 compiled_fn = disable(compiled_fn)
    599 counters["stats"]["unique_graphs"] += 1

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/utils.py:160, in dynamo_timed.<locals>.dynamo_timed_inner.<locals>.time_wrapper(*args, **kwargs)
    158     compilation_metrics[key] = []
    159 t0 = time.time()
--> 160 r = func(*args, **kwargs)
    161 time_spent = time.time() - t0
    162 # print(f"Dynamo timer: key={key}, latency={latency:.2f} sec")

File /opt/miniconda3/lib/python3.9/site-packages/torch/_dynamo/output_graph.py:683, in OutputGraph.call_user_compiler(self, gm)
    681 except Exception as e:
    682     compiled_fn = gm.forward
--> 683     raise BackendCompilerFailed(self.compiler_fn, e) from e
    684 return compiled_fn

BackendCompilerFailed: debug_wrapper raised CppCompileError: C++ compile error

Command:
g++ /var/folders/x7/scywf31j5znbyrc7sx9fyd900000gn/T/torchinductor_frankj/nv/cnvvffjhbnznyauxn6jlzj5qjcgxza6svl7hi5kmnnallfw3xfu4.cpp -shared -fPIC -Wall -std=c++17 -Wno-unused-variable -I/opt/miniconda3/lib/python3.9/site-packages/torch/include -I/opt/miniconda3/lib/python3.9/site-packages/torch/include/torch/csrc/api/include -I/opt/miniconda3/lib/python3.9/site-packages/torch/include/TH -I/opt/miniconda3/lib/python3.9/site-packages/torch/include/THC -I/opt/miniconda3/include/python3.9 -lgomp -march=native -O3 -ffast-math -fno-finite-math-only -fopenmp -D C10_USING_CUSTOM_GENERATED_MACROS -o/var/folders/x7/scywf31j5znbyrc7sx9fyd900000gn/T/torchinductor_frankj/nv/cnvvffjhbnznyauxn6jlzj5qjcgxza6svl7hi5kmnnallfw3xfu4.so

Output:
clang: error: the clang compiler does not support '-march=native'
clang: error: unsupported option '-fopenmp'
clang: error: unsupported option '-fopenmp'


Set torch._dynamo.config.verbose=True for more information


You can suppress this exception and fall back to eager by setting:
    torch._dynamo.config.suppress_errors = True
```

**Comment by crazy-JiangDongHua** [comment]
> @malfet this is probably the thing where the cpu integration doesn't work on osx right now. It's kind of shocking that they have cuda working on their system though lol

sorry, I remove cuda() call when I run the program in my macbook, and I forget to remove it and just copy sample program to the issue.

**Comment by ezyang** [comment]
Ok dupe of https://github.com/pytorch/pytorch/issues/90362

**Comment by crazy-JiangDongHua** [comment]
> Ok dupe of #90362

Ok, Thank you very much!

**Comment by malfet** [comment]
Ok, will prioritize pushing a fix next week

**Comment by albanD** [comment]
Closing as duplicate (and the other is already high pri)
